migration/table.go

```diff
@@ -52,6 +52,29 @@ func (m TableCreator) CreateTable(ctx context.Context, conn execution.QueryExece
 	return nil
 }
 
+// DropAndCreateTable generates DROP TABLE and CREATE TABLE definitions for the given table, and runs them on the given conn
+func (m TableCreator) DropAndCreateTable(ctx context.Context, conn execution.QueryExecer, t, parent *schema.Table) error {
```

Comment:
We're removing `TableCreator.CreateTable` in #190, I don't think we should add a new one here (that's also accessible from the outside) so I suggest moving this into an unexported fn under `provider/testing/`.